### PR TITLE
refactor: dedup file-copying code

### PR DESCRIPTION
**Pull request by ralfbrown**
This PR eliminates duplicate code to copy the contents of a file.

The first commit is safe to cherry-pick, as it just moves an existing function and elides the clone of that function from another file.

The second commit should get some testing before merge, as it involves actual refactoring of code and I haven't yet figured out what user actions are needed to trigger those four file-copy instances.

### CODE REVIEW

File: src/common/utility.c [hunk 1]
```diff
@@ -764,6 +764,82 @@ char *dt_util_format_exposure(const float exposuretime)
...
+
+  g_free(content);
+}
+
+void dt_copy_resource_file(const char *src, const char *dst)
+{
+  char share[PATH_MAX] = { 0 };
+  dt_loc_get_datadir(share, sizeof(share));
+  gchar *sourcefile = g_build_filename(share, src, NULL);
+  char *content = NULL;
```
Comment (TurboGit) [inline src/common/utility.c:820]: From there you can just call `dt_copy_file(sourcefile, dst)`.

**Review by TurboGit — CHANGES_REQUESTED** [review]
Some more refactoring possible.

**Comment by ralfbrown** [comment]
I had noticed that, but wanted to keep the untested code changes to a minimum.

**Review by TurboGit — APPROVED** [review]
Looks good now, thanks!

**Comment by TurboGit** [comment]
> I had noticed that, but wanted to keep the untested code changes to a minimum.

Sure, I have reviewed properly and the code is really identical to me. Thanks.

**Pull request merged**